stor/tests/test_integration_filemodes.py

```diff
@@ -0,0 +1,81 @@
+from nose.tools import raises
+from unittest import skipIf
+import six
+import stor
+import os
+
+byte_string = b'hey'
+string_string = 'hey'
+base_name = "test.txt"
+
+
+def generate_file(opener, test_filename, mode='b'):
+    if mode == "b":
+        fp = opener(test_filename, 'wb')
+        fp.write(byte_string)
+    else:
+        fp = opener(test_filename, 'w')
+        fp.write(string_string)
+
+
+class BaseTest(object):
+    __test__ = False
+
+    @property
+    def test_filename(self):
+        return self.basepath + "/" + base_name  # Yes I know
+
+    def test_write_bytes_to_binary(self):
+        fp = self.open(self.test_filename, mode='wb')
+        fp.write(byte_string)
+
+    @skipIf(not six.PY3, "Only tested on py3")
+    @raises(TypeError)
+    def test_write_string_to_binary(self):   # pragma: no cover
+        fp = self.open(self.test_filename, mode='wb')
+        fp.write(string_string)
+
+    @skipIf(not six.PY3, "Only tested on py3")
```

Comment:
you can just add the appropriate ones on to the base integration test (probably only need one or two of them) and then I'd suggest making it:

```
bytes_string = six.bytes_type(b'string')
string_string = six.text_type('string')
```

and then finishing the write.  I'd also like it if you could include an encoded unicode character in the bytes string and a decoded unicode character in the unicode string, just to uncover concatenation errors.